Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt

```diff
@@ -247,6 +241,19 @@ private fun processCLib(flavorName: String, cinteropArguments: CInteropArguments
 
     val outKtPkg = fqParts.joinToString(".")
 
+    val jarDirectory = def.config.j2objcJar
+    val j2objcIndexerResult: IndexerResult
+
+    if (language == Language.J2ObjC) {
+        val jarClassData = loadClassDataFromJar(JarFile(jarDirectory[0]))
+        val jarClassNodes = generateClassNodes(jarClassData)
+        j2objcIndexerResult =
+          IndexerResult(J2ObjCNativeIndex(jarClassNodes), CompilationWithPCH(emptyList<String>(), Language.J2ObjC))
+    } else {
+        j2objcIndexerResult =
+          IndexerResult(J2ObjCNativeIndex(emptyList<ObjCClass>()), CompilationWithPCH(emptyList<String>(), Language.J2ObjC))
+    }
+
```

Comment:
Why do this up here as opposed to as part of the if statement in indexResult?

val index:IndexerResult;
if (language == Language.J2ObjC) {
   ... this code ...
} else {
  nativeResult = buildNativeIndex(buildNativeLibrary(tool,def,cinteropArguments,imports), verbose)
}

or maybe pull this code out into a helper funtion and then we'd get

val index = if (language == Language.J2ObjC) buildJ2ObjCNativeIndex(buildNativeLibrary(tool,def,cinteropArguments,imports), verbose) else buildNativeIndex(buildNativeLibrary(tool,def,cinteropArguments,imports), verbose)

Reply:
Yeah, I ended up changing my code to the first option ( I don't exactly know what I was thinking at the time, I was struggling with the language selection for some reason)

I do like the second idea more so I implemented that now.